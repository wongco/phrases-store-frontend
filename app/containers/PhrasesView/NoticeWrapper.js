import styled from 'styled-components';

const NoticeWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
`;

export default NoticeWrapper;
