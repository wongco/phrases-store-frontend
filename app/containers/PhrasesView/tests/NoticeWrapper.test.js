import React from 'react';
import { shallow, mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import toJson from 'enzyme-to-json';
import NoticeWrapper from '../NoticeWrapper';

describe('<NoticeWrapper />', () => {
  it('renders without crashing', () => {
    shallow(<NoticeWrapper />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<NoticeWrapper />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const id = 'testId';
    const wrapper = mount(<NoticeWrapper id={id} />);
    const renderedComponent = enzymeFind(wrapper, NoticeWrapper);
    expect(renderedComponent.prop('id')).toBe('testId');
  });

  it('should render its text', () => {
    const children = 'Text';
    const wrapper = mount(<NoticeWrapper>{children}</NoticeWrapper>);
    const renderedComponent = enzymeFind(wrapper, NoticeWrapper);
    expect(renderedComponent.contains(children)).toBe(true);
  });
});
