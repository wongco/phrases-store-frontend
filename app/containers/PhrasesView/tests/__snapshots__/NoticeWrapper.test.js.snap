// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoticeWrapper /> matches snapshot for default criteria 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "NoticeWrapper-sc-1dc1n48-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;justify-content:center;align-items:center;width:100%;",
        ],
      },
      "displayName": "NoticeWrapper",
      "render": [Function],
      "styledComponentId": "NoticeWrapper-sc-1dc1n48-0",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
